### PR TITLE
Feautre 64 create popup

### PR DESCRIPTION
**Pull request by ghost**
I've done the task. Perhaps I inattentively read the condition and did the task from "Map-70".

### CODE REVIEW

File: backend/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,1044 @@
+{
+  "name": "backend",
+  "version": "1.0.0",
+  "lockfileVersion": 1,
+  "requires": true,
+  "dependencies": {
```
Comment (steadyGuy) [inline backend/package-lock.json]: Remove this file (package-lock) as we don't use "npm" in our project

File: frontend/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,12510 @@
+{
+  "name": "frontend",
```
Comment (steadyGuy) [inline frontend/package-lock.json]: remove this file too

File: frontend/src/components/modals/pointPopup/PointPopup.tsx [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+  FormControl,
+  Button,
+  List,
+  ListItem,
+  Divider,
+  ListItemText,
+  ListItemAvatar
+} from '@mui/material';
+
+import './pointPopup.scss';
```
Comment (steadyGuy) [inline frontend/src/components/modals/pointPopup/PointPopup.tsx]: we don't use scss

File: frontend/src/components/modals/pointPopup/PointPopup.tsx [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+  const { expand, ...other } = props;
+  /* eslint-disable-next-line react/jsx-props-no-spreading */
+  return <IconButton {...other} />;
+})(({ theme, expand }) => ({
+  transform: !expand ? 'rotate(0deg)' : 'rotate(180deg)',
+  marginLeft: 'auto',
+  transition: theme.transitions.create('transform', {
+    duration: theme.transitions.duration.shortest
+  })
+}));
```
Comment (steadyGuy) [inline frontend/src/components/modals/pointPopup/PointPopup.tsx]: You can place it inside _styles.ts_. But it's up to you.

File: frontend/src/components/modals/pointPopup/pointPopup.scss [hunk 5]
```diff
@@ -0,0 +1,29 @@
+* {
+  padding: 0;
+  margin: 0;
+  border: 0;
+  box-sizing: border-box;
+}
```
Comment (steadyGuy) [inline frontend/src/components/modals/pointPopup/pointPopup.scss]: We don't use scss. Maybe try to look at this https://emotion.sh/docs/globals. And try to replace global scss via global emotion component

**Review by steadyGuy — CHANGES_REQUESTED** [review]

File: backend/package-lock.json (hunk 1 above)
Comment (ghost) [reply backend/package-lock.json]: Ok, no problem

File: frontend/package-lock.json (hunk 2 above)
Comment (ghost) [reply frontend/package-lock.json]: Ok, no problem

File: frontend/src/components/modals/pointPopup/PointPopup.tsx (hunk 3 above)
Comment (ghost) [reply frontend/src/components/modals/pointPopup/PointPopup.tsx]: I didn't know we don't use scss. I'll fix it. But it seems to me that this gives additional opportunities for more flexible customization.

File: frontend/src/components/modals/pointPopup/PointPopup.tsx (hunk 4 above)
Comment (ghost) [reply frontend/src/components/modals/pointPopup/PointPopup.tsx]: I'll fix it.

File: frontend/src/components/modals/pointPopup/pointPopup.scss (hunk 5 above)
Comment (ananchenko346) [reply frontend/src/components/modals/pointPopup/pointPopup.scss:6]: This is a general style reset for the entire application. I'm not sure if <CssBaseline /> handles this correctly.

Comment (ghost) [reply frontend/src/components/modals/pointPopup/pointPopup.scss]: This is a general style reset for the entire application. I'm not sure if CssBaseline handles this correctly.

File: frontend/src/components/modals/pointPopup/PointPopup.tsx (hunk 3 above)
Comment (steadyGuy) [reply frontend/src/components/modals/pointPopup/PointPopup.tsx]: > I didn't know we don't use scss. I'll fix it. But it seems to me that this gives additional opportunities for more flexible customization.

Maybe) But it's violates code convention too.

File: frontend/src/components/modals/pointPopup/pointPopup.scss (hunk 5 above)
Comment (steadyGuy) [reply frontend/src/components/modals/pointPopup/pointPopup.scss]: > This is a general style reset for the entire application. I'm not sure if CssBaseline handles this correctly.

It's okey. I think Global component should work to reset

File: frontend/src/components/modals/pointPopup/PointPopup.tsx [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+
+// It can be reused in different components
+// When the rest of the functionality is available, this component will be assigned to a specific and each point on the map
+// Change position a photo, description, buttons, etc. you can without problems
+// Now it works on stubs
+
+interface Props {
+  active: boolean;
+  setActive: Function;
+  children: Array<any>;
```
Comment (steadyGuy) [inline frontend/src/components/modals/pointPopup/PointPopup.tsx]: Why any?

**Review by steadyGuy — APPROVED** [review]

File: frontend/src/components/modals/pointPopup/pointPopup.scss (hunk 5 above)
Comment (steadyGuy) [reply frontend/src/components/modals/pointPopup/pointPopup.scss]: scss still here.
https://stackoverflow.com/a/69218520

**Review by steadyGuy — CHANGES_REQUESTED** [review]

File: frontend/src/components/modals/pointPopup/PointPopup.tsx (hunk 6 above)
Comment (ghost) [reply frontend/src/components/modals/pointPopup/PointPopup.tsx]: What should be there?
It's like a stub.
Data can changes many times.

Comment (steadyGuy) [reply frontend/src/components/modals/pointPopup/PointPopup.tsx]: create a global interface inside types.ts file for the object you pass in
image

Comment (steadyGuy) [reply frontend/src/components/modals/pointPopup/PointPopup.tsx]: export interface IPointBody {
locationName: string;
avatarMini: string;
name: string;
...
}

Comment (steadyGuy) [reply frontend/src/components/modals/pointPopup/PointPopup.tsx]: create types.ts inside src folder

Comment (ghost) [reply frontend/src/components/modals/pointPopup/PointPopup.tsx]: This data is not final.
There may be a completely different structure and data types.

Comment (steadyGuy) [reply frontend/src/components/modals/pointPopup/PointPopup.tsx]: Then interface would change

**Review by vasylneshchadimov — COMMENTED** [review]
Why you write code in Map component?
I think will beter if you put  all code in individualcomponent.

**Comment by ghost** [comment]
This is test data but I have already changed it.

File: frontend/src/components/modals/pointPopup/pointPopup.scss (hunk 5 above)
Comment (ghost) [reply frontend/src/components/modals/pointPopup/pointPopup.scss]: I have reseted all styles via GlobalStyle

**Review by vasylneshchadimov — COMMENTED** [review]
Popup doesn't open correctly. Top of Popup mast be under header, and doesn't overlay it

**Comment by ghost** [comment]
Well, you did the layout. I just put my pop up in your pour. Then you need to rework your layout. You have to to make a normal shell so that it opens correctly. I did my job and put it together. My job was to make a popup, I made it identical to the layout in Figma.

**Comment by vasylneshchadimov** [comment]
Vlad, pull develop brench, and watch how my popup open

**Comment by ghost** [comment]
And in general, let's start with the fact that first I had to make a popup and then you had to make a layout for this popup. But I made concessions in this moment. After that I just put my popup in your crooked layout. The fact that LAYOUT opens and works not as it should is your task to fix it, not mine.

**Comment by ghost** [comment]
It fixed and wrong

File: frontend/src/components/Map/index.tsx [hunk 7]
```diff
@@ -12,14 +13,17 @@ function Map() {
...
-      zoom={9}
-      style={{ height: '100vh' }}
-    >
-      <TileLayer url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" />
-    </MapContainer>
+    <Box>
+      <MapContainer
+        center={[50.447731, 30.542721]}
+        zoom={9}
+        style={{ height: '100vh' }}
```
Comment (steadyGuy) [inline frontend/src/components/Map/index.tsx:22]: try to not use inline styles (especially if it's not a __sx__). In your case, you can style parent <Box> and inside this box, add style for his children (or whatever other implementation).

File: frontend/src/components/PointPopup/PointPopup.tsx [hunk 8]
```diff
@@ -0,0 +1,418 @@
...
+  ListItemAvatar
+} from '@mui/material';
+
+// It can be reused in different components
+// When the rest of the functionality is available, this component will be assigned to a specific and each point on the map
+// Change position a photo, description, buttons, etc. you can without problems
+// Now it works on stubs
+
+interface Props {
+  children: Array<any>;
```
Comment (steadyGuy) [inline frontend/src/components/PointPopup/PointPopup.tsx:35]: okay, if you think that now this is acceptable, than leave it as it is

File: frontend/src/components/PointPopup/PointPopup.tsx [hunk 9]
```diff
@@ -0,0 +1,418 @@
...
+}
+
+interface ExpandMoreProps extends IconButtonProps {
+  expand: boolean;
+}
+
+const ExpandMore = styled((props: ExpandMoreProps) => {
```
Comment (steadyGuy) [inline frontend/src/components/PointPopup/PointPopup.tsx:42]: you can create styles.ts, as you do for Map

File: frontend/src/components/PointPopup/PointPopup.tsx [hunk 10]
```diff
@@ -0,0 +1,418 @@
...
+                  <Typography>{children[0].name}</Typography>
+
+                  <Typography>{children[0].dateOfPublic}</Typography>
+                </Typography>
+                <Typography
+                  variant="subtitle1"
+                  sx={{
+                    mt: 5,
+                    p: 3,
+                    border: '1px solid grey '
```
Comment (steadyGuy) [inline frontend/src/components/PointPopup/PointPopup.tsx:181]: well, better get it from the theme

**Review by steadyGuy — APPROVED** [review]

File: frontend/src/components/Map/index.tsx (hunk 7 above)
Comment (ghost) [reply frontend/src/components/Map/index.tsx:22]: I didn't write it. This code was already there before me.

**Review by vasylneshchadimov — APPROVED** [review]

**Review by ghost — COMMENTED** [review]
To the 'full popup'. 
Vasyl, I'm very sorry but this is not right. In this case popup doesn't work correctly. When the popup leaves in your case there is no bottom arrow that should open the additional functionality of the popup. In your case the popup will be without a drop-down list.

**Review by ghost — COMMENTED** [review]
Vasyl, I'm so sorry but this is not right. In this case popup doesn't work correctly. When the popup leaves in your case there is no bottom arrow that should open the additional functionality of the popup. In your case the popup will be without a drop-down list.

**Pull request merged**